internal/services/nginx/nginx_configuration_resource.go

```diff
@@ -0,0 +1,318 @@
+package nginx
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/nginx/2022-08-01/nginxconfiguration"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/nginx/2022-08-01/nginxdeployment"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+const defaultConfigurationName = "default"
+
+type ConfigFile struct {
+	Content     string `tfschema:"content"`
+	VirtualPath string `tfschema:"virtual_path"`
+}
+
+type ProtectedFile struct {
+	Content     string `tfschema:"content"`
+	VirtualPath string `tfschema:"virtual_path"`
+}
+
+type ConfigurationModel struct {
+	Name              string          `tfschema:"name"` // always default
+	NginxDeploymentId string          `tfschema:"nginx_deployment_id"`
+	ConfigFile        []ConfigFile    `tfschema:"config_file"`
+	ProtectedFile     []ProtectedFile `tfschema:"protected_file"`
+	PackageData       string          `tfschema:"package_data"`
+	RootFile          string          `tfschema:"root_file"`
+}
+
+// ToSDKModel used in both Create and Update
+func (c ConfigurationModel) ToSDKModel() nginxconfiguration.NginxConfiguration {
```

Comment:
this approach means that `ignore_changes` won't work, the Create and Update should use different methods (and be inlined within each) - so that the Update can use `d.HasChanges("field")` as appropriate to update the config